sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -473,12 +475,23 @@ export class EventHubSender extends LinkEntity {
             }
           };
 
+          const onAborted: any = () => {
+            removeListeners();
+            const desc: string =
+              `[${this._context.connectionId}] The send operation on the Sender "${this.name}" with ` +
+              `address "${this.address}" has been cancelled by the user.`;
+            log.error(desc);
+            throw new Error(desc);
+          };
           onAccepted = (context: EventContext) => {
             // Since we will be adding listener for accepted and rejected event every time
             // we send a message, we need to remove listener for both the events.
             // This will ensure duplicate listeners are not added for the same event.
             removeListeners();
             log.sender("[%s] Sender '%s', got event accepted.", this._context.connectionId, this.name);
+            if (aborter) {
+              aborter.removeEventListener("abort", onAborted);
+            }
```

Comment:
We should remove the listener for aborter in 4 places where we resolve/reject promises.
So, we should move this inside `removeListeners()`